src/integrationTest/java/org/opensearch/security/SecurityConfigurationTests.java

```diff
@@ -0,0 +1,223 @@
+	public static final String INTERNAL_USERS_RESOURCE = "/_plugins/_security/api/internalusers/";
+	public static final String ID_1 = "one";
+	public static final String PROHIBITED_INDEX = "prohibited";
+	public static final String ID_2 = "two";
+
+	@ClassRule
+	public static LocalCluster cluster = new LocalCluster.Builder()
+		.clusterManager(ClusterManager.THREE_CLUSTER_MANAGERS)
+		.authc(AUTHC_HTTPBASIC_INTERNAL).users(USER_ADMIN, LIMITED_USER).anonymousAuth(false)
+		.nodeSettings(Map.of(SECURITY_RESTAPI_ROLES_ENABLED, List.of("user_" + USER_ADMIN.getName()  +"__" + ALL_ACCESS.getName()),
+			SECURITY_BACKGROUND_INIT_IF_SECURITYINDEX_NOT_EXIST, true))
+		.build();
+
+	@Rule
+	public TemporaryFolder configurationDirectory = new TemporaryFolder();
+
+	@BeforeClass
+	public static void initData() {
+		try(Client client = cluster.getInternalNodeClient()){
+			client.prepareIndex(LIMITED_USER_INDEX).setId(ID_1).setRefreshPolicy(IMMEDIATE).setSource("foo", "bar").get();
+			client.prepareIndex(PROHIBITED_INDEX).setId(ID_2).setRefreshPolicy(IMMEDIATE).setSource("three", "four").get();
+		}
+	}
+
+	@Test
+	public void shouldCreateUserViaRestApi_success() {
+		try(TestRestClient client = cluster.getRestClient(USER_ADMIN)) {
+			HttpResponse httpResponse = client.putJson(INTERNAL_USERS_RESOURCE + ADDITIONAL_USER_1, String.format(CREATE_USER_BODY,
+				ADDITIONAL_PASSWORD_1));
+
+			assertThat(httpResponse, notNullValue());
+			assertThat(httpResponse.getStatusCode(), equalTo(201));
+		}
+		try(TestRestClient client = cluster.getRestClient(USER_ADMIN)) {
+			client.assertCorrectCredentials();
+		}
+		try(TestRestClient client = cluster.getRestClient(ADDITIONAL_USER_1, ADDITIONAL_PASSWORD_1)) {
+			client.assertCorrectCredentials();
+		}
+	}
+
+	@Test
+	public void shouldCreateUserViaRestApi_failure() {
+		try(TestRestClient client = cluster.getRestClient(LIMITED_USER)) {
+			HttpResponse httpResponse = client.putJson(INTERNAL_USERS_RESOURCE + ADDITIONAL_USER_1, String.format(CREATE_USER_BODY,
+				ADDITIONAL_PASSWORD_1));
+
+			assertThat(httpResponse, notNullValue());
+			httpResponse.assertStatusCode(403);
+		}
+	}
+
+	@Test
+	public void shouldAuthenticateAsAdminWithCertificate_positive() {
+		try(TestRestClient client = cluster.getRestClient(cluster.getAdminCertificate())) {
+			HttpResponse httpResponse = client.get("/_plugins/_security/whoami");
+
+			assertThat(httpResponse, notNullValue());
+			httpResponse.assertStatusCode(200);
+			assertThat(httpResponse.getTextFromJsonBody("/is_admin"), equalTo("true"));
+		}
+	}
+
+	@Test
+	public void shouldAuthenticateAsAdminWithCertificate_negativeSelfSignedCertificate() {
+		TestCertificates testCertificates = cluster.getTestCertificates();
+		try(TestRestClient client = cluster.getRestClient(testCertificates.createSelfSignedCertificate("CN=bond"))) {
+			HttpResponse httpResponse = client.get("/_plugins/_security/whoami");
+
+			assertThat(httpResponse, notNullValue());
```

Comment:
Nit: unless sometimes we expect these responses to be null, these checks are not very useful.  What would it mean if this assertion failed, what action should be taken?

Reply:
done